src/actors/apache_scanner/apache_scanner.py

```diff
@@ -0,0 +1,137 @@
+                    yield key + "=" + table[key]
+        if 'properties' in table:
+            for generator in searchblock(table['properties'], mykey, myvalue, block):
+                yield generator
+
+
+def getblock(tablelist, mykey, myvalue=None, block=True, propitem=0):
+    """ Accumulates matched data and returns them for output.
+
+        Arguments:
+            block: if False, will pass block=False to searchblock
+            propitem: specify from which index in 'properties' to search
+
+        Returns:
+            Extracted data in form of dictionary.
+    """
+    out = {}
+    if block is True:
+        for dic in searchblock(tablelist, mykey, myvalue, block):
+            if dic[mykey] not in out:
+                out[dic[mykey]] = [dic['properties'][propitem]['value']]
+            else:
+                out[dic[mykey]].append(dic['properties'][propitem]['value'])
+    else:
+        for line in searchblock(tablelist, mykey, myvalue, block):
+            key, value = line.split("=")
+            if key not in out:
+                out[key] = [value]
+            else:
+                out[key].append(value)
+    return out
+
+
+def printall(tablelist):
+    """ Prints all occurrences of the specified key-value pairs.
+
+    Tuple @searches contains information about what to extract and how it will
+    be extracted (see @block and propitem@ arguments to getblock() function.
+
+    Args:
+        tablelist: 'aug_httpd' lens data from the augeas actor output
+
+    Returns:
+        json formatted extracted data from the augeas aug_httpd lens
+    """
+    searches = (
+        # Arguments for the getblock function.
+        # mykey, myvalue=None, block=True, propitem=0
+        ("value", "IncludeOptional"),
+        ("value", "User"),
+        ("value", "Group"),
+        ("value", "Mutex"),
+        ("value", "PidFile"),
+        ("value", "Listen"),
+        ("value", "TypesConfig"),
+        ("value", "CacheRoot"),
+        ("value", "MIMEMagicFile"),
+        ("value", "ScriptSock"),
+        ("value", "SSLCertificateKeyFile"),
+        ("value", "SSLCertificateFile"),
+        ("value", "CustomLog"),
+        ("value", "ErrorLog"),
+        ("value", "Include"),
+        ("name", "VirtualHost"),
+        ("name", "Directory"),
+        ("value", "DocumentRoot"),
+        ("absolute_path", None, False),
+        ("value", "LoadModule", True, 1),
+    )
+
+    out = {}
+    for args in searches:
+        result_table = getblock(tablelist, *args)
+        for key in result_table:
+            out[key] = result_table[key]
+    return {"apache_scanner": [out]}
+
+
+if __name__ == "__main__":
+    AUGEAS_OUTPUT = json.load(sys.stdin)
+    try:
+        # We are interested only in the aug_httpd lens from the augeas actor
+        #  output. We can drop every other augeas lens.
+        TABLELIST = AUGEAS_OUTPUT['aug_httpd']
+    except KeyError:
+        raise KeyError("Could not get lens 'aug_httpd' from the augeas actor " \
```

Comment:
Hmm i think you could use the same exception instance but add message to that. I think it's better to keep original message there.